internal/sftp/goclient.go

```diff
@@ -0,0 +1,84 @@
+package sftp
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/pkg/sftp"
+	"golang.org/x/crypto/ssh"
+)
+
+// GoClient implements the SFTP service using native Go SSH and SFTP packages.
+type GoClient struct {
```

Comment:
Perhaps one mention I would make there is that this client is not thread-safe, e.g. to concurrent actors wouldn't be able to perform uploads in the way this is implemented, which I think is okay but it's something commonly documented from what I see in other projects.

Reply:
Ooooh, I see - if `dial()` is called a second time it overwrites the `c.ssh` and `c.sftp` clients. :thinking:

Reply:
Assuming `ssh.Client` and `sftp.Client` are thread safe, In `dial()` I think I could just check if each client is not nil before creating a new one.  Do you think that would be sufficient for thread safety?

Reply:
I believe you're right!

Reply:
Making this thread-safe turned out to be harder than I expected, so I've just added a comment that Upload is not thread safe for now.